src/ploigos_step_runner/step_implementers/generate_evidence/generate_evidence.py

```diff
@@ -0,0 +1,237 @@
+        """Runs the step implemented by this StepImplementer.
+
+        Returns
+        -------
+        StepResult
+            Object containing the dictionary results of this step.
+        """
+        step_result = StepResult.from_step_implementer(self)
+
+        #Gather all evidence from steps/sub steps
+        evidence = self.__gather_evidence()
+
+        #If no evidence is found then return a step result reflecting that status.
+        if not evidence:
+            step_result.add_artifact(
+                name='result-generate-evidence',
+                value='No evidence to generate.',
+                description='Evidence from all previously run steps.'
+            )
+            step_result.message = "No evidence generated from previously run steps"
+            return step_result
+
+        org = self.get_value('organization')
+        app = self.get_value('application-name')
+        service = self.get_value('service-name')
+        version = self.get_value('version')
+
+        result_evidence_name = f"{org}-{app}-{service}-{version}-evidence.json"
+
+        work_dir = self.work_dir_path
+
+        evidence_path = os.path.join(work_dir, result_evidence_name)
+
+        #Format into json and write out to file in working directory
+        with open(evidence_path, "w") as evidence_file:
+            json.dump(evidence, evidence_file, indent=4)
+
+        evidence_destination_url = self.get_value('evidence-destination-url')
+        evidence_destination_uri = f"{evidence_destination_url}/" \
+                f"{org}/{app}/{service}/{result_evidence_name}"
+
+        #Upload file to datastore
+        if evidence_file and evidence_destination_url:
+            try:
+                upload_result = upload_file(
+                    file_path=evidence_path,
+                    destination_uri=evidence_destination_uri,
+                    username=self.get_value('evidence-destination-username'),
+                    password=self.get_value('evidence-destination-password')
+                )
+                step_result.add_artifact(
+                    name='results-evidence-upload-results',
+                    description='Results of uploading the evidence JSON file ' \
+                        'to the given destination.',
+                    value=upload_result
+                )
+                step_result.add_artifact(
+                name='evidence-uri',
+                description='URI of the uploaded results archive.',
+                value=evidence_destination_uri
+                )
+            except RuntimeError as error:
+                step_result.success = False
+                step_result.message = str(error)
+                return step_result
+
+        step_result.success = True
+        step_result.message = 'Evidence successfully packaged ' \
+            'in JSON file and uploaded to data store.'
+
+        step_result.add_artifact(
+            name='result-generate-evidence-path',
```

Comment:
simply this name to `evidence-path` or even just `evidence`. dealer choice. update the class doc too.

Reply:
Fixed